addons/stock/models/stock_package.py

```diff
@@ -13,44 +13,98 @@ class StockPackage(models.Model):
     _name = 'stock.package'
     _description = "Packages"
     _order = 'name'
+    _parent_name = 'parent_package_id'
+    _parent_store = True
+    _rec_name = 'complete_name'
 
     name = fields.Char(
         'Package Reference', copy=False, index='trigram', required=True,
         default=lambda self: self.env['ir.sequence'].next_by_code('stock.package') or self.env._('Unknown Pack'))
+    complete_name = fields.Char("Full Package Name", compute='_compute_complete_name', recursive=True, store=True)
+    is_named_by_seq = fields.Boolean("Named by Sequence", compute="_compute_is_named_by_seq")
     quant_ids = fields.One2many('stock.quant', 'package_id', 'Bulk Content', readonly=True,
         domain=['|', ('quantity', '!=', 0), ('reserved_quantity', '!=', 0)])
+    contained_quant_ids = fields.One2many('stock.quant', compute="_compute_contained_quant_ids", search="_search_contained_quant_ids")
     package_type_id = fields.Many2one(
         'stock.package.type', 'Package Type', index=True)
     location_id = fields.Many2one(
         'stock.location', 'Location', compute='_compute_package_info',
         index=True, readonly=False, store=True)
+    location_dest_ids = fields.Many2many('stock.location', 'Destination locations', compute='_compute_location_dest_ids')
```

Comment:
Why is it a m2m? shouldn't be a m2o?

Reply:
As discussed: Because a single package *could* be assigned as destination package in different pickings that have different dests, as long as none of them is validated and the package is currently empty.
Will change it to a m2o though, as this represents a corner case anyway that won't happen 99% of the time.